examples/threadExample/src/main/java/io/realm/examples/threads/AsyncQueryFragment.java

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright 2015 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.examples.threads;
+
+import android.content.Context;
+import android.os.Bundle;
+import android.support.v4.app.Fragment;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.BaseAdapter;
+import android.widget.ListView;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import java.util.Collections;
+import java.util.List;
+
+import io.realm.Realm;
+import io.realm.RealmChangeListener;
+import io.realm.RealmResults;
+import io.realm.examples.threads.model.Dot;
+
+/**
+ * This fragment demonstrates how you can perform asynchronous queries with Realm
+ */
+public class AsyncQueryFragment extends Fragment implements View.OnClickListener, RealmChangeListener {
+    private Realm realm;
+    private DotAdapter dotAdapter;
+    private RealmResults<Dot> allSortedDots;
+    private Realm.Request asyncTransaction;
```

Comment:
There is something about the naming here that bothers me, it sounds wrong. `Realm.Request` is used for both async queries and write transactions. Retrofit can get away with calling anything `Request` as it fits the HTTP model, I don't think it fits that well for queries/transactions. Also it will be problematic to have this class as a subclass to Realm when DynamicRealm enters the picture.

I suggest:
1) Move to to a top-level class to better support both Realm/DynamicRealm
2) Rename to something like `RealmAsyncTask`?